Options/Applicative/Builder/Internal.hs

```diff
@@ -82,6 +86,22 @@ instance HasMetavar ArgumentFields where
 instance HasMetavar CommandFields where
   hasMetavarDummy _ = ()
 
+class HasGroup f where
+  setGroup :: String -> f a -> f a
+  getGroup :: f a -> Maybe String
+instance HasGroup OptionFields where
+  setGroup n fields = fields { optGroup = Just n }
+  getGroup = optGroup
+instance HasGroup CommandFields where
```

Comment:
Having an instance for `CommandGroup` will almost certainly cause confusion. If I use the same name for a command and an option do they appear in the same place?

They should either be separate concepts or the same, but not both.